### PR TITLE
first try

### PR DESCRIPTION
**Pull request by Nikolya777777777777**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+            }
+        }
+        if (loginLength.length < 6) {
+            throw new NoValidDataException("Login should have at least 6 characters");
+        }
+        if (passwordLength.length < 6) {
+            throw new NoValidDataException("Password should have at least 6 characters");
+        }
+        if (user.getAge() < 18) {
+            throw new NoValidDataException("User need to be at least 18 years old");
```
[the remark this answers is not in the document]
Comment (okuzan) [reply src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:28]: Consider moving the hardcoded values like '6' for minimum login and password length, and '18' for minimum age to constant fields with descriptive names. This will make your code more readable and easier to maintain.

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        char []loginLength = user.getLogin().toCharArray();
+        char []passwordLength = user.getPassword().toCharArray();
```
[the remark this answers is not in the document]
Comment (okuzan) [reply src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:15]: It's unnecessary to convert login and password to char arrays just to check their lengths. You can directly use user.getLogin().length() and user.getPassword().length().

File: src/main/java/core/basesyntax/Main/Main.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.Main;
+
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+
+public class Main {
+    public static void main(String[] args) {
+        RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+        User user1 = new User(10l, "Mykola", "1234567", 21);
```
[the remark this answers is not in the document]
Comment (okuzan) [reply src/main/java/core/basesyntax/Main/Main.java:9]: The 'main' method is not needed for testing purposes as per the checklist: 'You don't need the main method to see how your solution works, you have tests for this purpose.' Consider removing this method.

File: src/main/java/core/basesyntax/Main/Main.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.Main;
```
[the remark this answers is not in the document]
Comment (okuzan) [reply src/main/java/core/basesyntax/Main/Main.java:1]: you don't need a package for main
and packages should be in lowercase

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
build failed, fix it first

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        User user1 = new User("Mykola", "1234567", 85);
+        Storage.people.add(user1);
+        User user2 = new User("Mykola", "1234567", 85);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user2);
+        });
+    }
+
+    @Test
+    void check_IfMethodCanAddNewUser_Ok() {
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:42]: use the same method name format 
```suggestion
    void check_ifMethodCanAddNewUser_ok() {
```

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    void register_loginIsLessThanMinLength_notOk() {
+        User user = new User("Myk", "12345568979", 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_ageIsLessThanMinLength_notOk() {
+        User user = new User("Mykola758", "12345568979", 15);
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:67]: better pass edge cases as input parameter 
```suggestion
        User user = new User("Mykola758", "12345568979", 17);
```

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    void register_loginIsLessThanMinLength_notOk() {
+        User user = new User("Myk", "12345568979", 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_ageIsLessThanMinLength_notOk() {
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:66]:
```suggestion
    void register_ageIsLessThanMinAge_notOk() {
```

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        User user = new User("Mykola8908", "123456789", -13);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:104]:
```suggestion
```

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 9]
```diff
@@ -2,13 +2,32 @@
...
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("You can not add user without any parameters");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with such login already exists");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < MINIMAL_AMOUNT_OF_SYMBOLS) {
+            throw new RegistrationException("Login should have at least 6 characters");
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:22]: better pass constant to message 
```suggestion
            throw new RegistrationException("Login should have at least " + MINIMAL_AMOUNT_OF_SYMBOLS + '' characters");
```
when you choose to change this requirement, you will only need to change the value of the constant, the messages will be valid (fix other messages)

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    void register_DuplicateUsers_NotOk() {
+        User user1 = new User("Mykola", "1234567", 85);
+        Storage.people.add(user1);
+        User user2 = new User("Mykola", "1234567", 85);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user2);
+        });
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/RegistrationServiceImplTest.java:38]: It will also be good to check the text of the exception message since this exception also occurs with other violations of the requirements for input parameters

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java (hunk 5 above)
Comment (ahoienko) [reply src/test/java/core/basesyntax/RegistrationServiceImplTest.java:42]: Not resolved

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java (hunk 6 above)
Comment (ahoienko) [reply src/test/java/core/basesyntax/RegistrationServiceImplTest.java:67]: Not resolved

File: src/test/java/core/basesyntax/RegistrationServiceImplTest.java (hunk 7 above)
Comment (ahoienko) [reply src/test/java/core/basesyntax/RegistrationServiceImplTest.java:66]: Not resolved

**Review by ahoienko — CHANGES_REQUESTED** [review]
Please read comments and resolved them

**Comment by Nikolya777777777777** [comment]
afe8028a161605a3e692ecab0d6cf4392ccfe55b  check this commit, because I changed names of methods

**Review by ahoienko — APPROVED** [review]
Good job!
